src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2266,9 +2266,11 @@ public static function unguarded(callable $callback)
 
         static::unguard();
 
-        $result = $callback();
-
-        static::reguard();
+        try {
+            $result = $callback();
+        } finally {
+            static::reguard();
+        }
 
         return $result;
```

Comment:
that is true, but writing the return inside the try like that is definitely preferred please